addons/event/models/event_mail.py

```diff
@@ -146,6 +150,14 @@ def _compute_mail_state(self):
             else:
                 scheduler.mail_state = 'running'
 
+    @api.constrains('notification_type', 'template_ref')
+    def _check_template_ref_model(self):
+        for record in self:
+            result = self._prepare_check_template_ref_model_values()
+            for ntype, model in result:
+                if record.notification_type == ntype and record.template_ref._name != model:
+                    raise ValidationError(_('The template which is referenced should be coming from %s model.') % model)
```

Comment:
Maybe use named parameter, like '... coming from %(model_name)s model" ? Easier for translations.